src/domain/usecases/user-monitoring/permission-fixer/__tests__/RunUserPermissionUseCase.specs.ts

```diff
@@ -0,0 +1,336 @@
+            name: "userusername",
+        });
+        expect(result.rolesReport?.invalidUsersCount).toEqual(0);
+        expect(result.message).toEqual("OK");
+    });
+
+    it("Should push fixed usergroup if the user has no template usergroups", async () => {
+        const useCase = givenUseCaseUserWithoutTemplateUserGroup();
+
+        const result = await useCase.execute();
+        expect(result.groupsReport?.invalidUsersCount).toEqual(1);
+        expect(result.groupsReport?.listOfAffectedUsers).toEqual([
+            {
+                id: "CHbcHcmgoZ5",
+                name: "userusername",
+            },
+        ]);
+        expect(result.rolesReport?.invalidUsersCount).toEqual(0);
+        expect(result.rolesReport?.listOfAffectedUsers).toEqual([]);
+        expect(result.message).toEqual("OK");
+    });
+
+    it("Should apply minimal control template group if a user dont have any control template user group", async () => {
+        const useCase = givenUseCaseUserWithoutTemplateUserGroup();
+
+        const result = await useCase.execute();
+        expect(result.groupsReport?.invalidUsersCount).toEqual(1);
+        expect(result.groupsReport?.listOfAffectedUsers[0]).toEqual({
+            id: "CHbcHcmgoZ5",
+            name: "userusername",
+        });
+        expect(result.rolesReport?.invalidUsersCount).toEqual(0);
+        expect(result.message).toEqual("OK");
+    });
+});
+
+beforeEach(() => {
+    clonedFakeUserWithoutGroup = copyObject(fakeUserWithoutUserGroup);
+    clonedValidUser = copyObject(fakeValidUser);
+    clonedInvalidUser = copyObject(fakeInvalidUser);
+    clonedFakeUserWithoutGroup = copyObject(fakeUserWithoutUserGroup);
+    configWithWrongMinimalGroup = copyObject(baseMetadataConfig);
+    configWithWrongMinimalGroup.minimalGroup = {
+        id: "IKpEgoQ4S03",
+        name: "WrongMinimal group uid",
+    };
+    configWithWrongMinimalGroup.permissionFixerConfig.forceMinimalGroupForUsersWithoutGroup = false;
+    configWithWrongMinimalGroup.permissionFixerConfig.pushFixedUserGroups = true;
+
+    configWithUserExcluded = copyObject(baseMetadataConfig);
+    configWithUserExcluded.excludedUsers = [copyObject(fakeValidUser)];
+    configThrowInvalidUsergroupException = copyObject(baseMetadataConfig);
+    clonedTemplateAuthorities = copyObject(permissionFixerTemplateGroupsExtended);
+    const permissionFixerInvalidTemplateGroupsExtended: PermissionFixerTemplateGroupExtended = copyObject(
+        permissionFixerTemplateGroupsExtended[0]
+    );
+    permissionFixerInvalidTemplateGroupsExtended.group = { id: "notValidGroup", name: "not valid group" };
+    clonedInvalidTemplateAuthorities = copyObject(permissionFixerInvalidTemplateGroupsExtended);
+});
+
+function getMetadataConfig(): PermissionFixerMetadataConfig {
+    return baseMetadataConfig;
+}
+
+function getMetataConfiThrowInvalidUsergroupException(): PermissionFixerMetadataConfig {
+    return configThrowInvalidUsergroupException;
```

Comment:
This function is not necessary and generate noise, you can use the variable direcly